overlay.nix

```diff
@@ -0,0 +1,13 @@
+self: super: {
+
+  shab = super.callPackage ./shab.nix {};
+
+  runShab = name: parameters: shabScript:
```

Comment:
could the name be the `builtins.baseNameOF shabScript`?

For composition I think it makes sense to have the template file first and parameters second.

Reply:
I've decided to expose the leakiness of `parameters` instead.  Maybe we could sanitize the basename and set that as a default? Doesn't really work for `runShabText` though :(